### PR TITLE
feat(graph): passthrough input types to invoke/stream

### LINKED ISSUE #628: Typesafe graph.invoke?
As far as I can tell, is there not a way to make the invoke method typesafe with the options argument `await graph.invoke({ key1: value })`?

## Expected Output
```ts
const StateAnnotation = Annotation.Root({
	messages: Annotation<string[]>({
		reducer: (x, y) => x.concat(y),
	}),
})

// Define a new graph
const workflow = new StateGraph(StateAnnotation)

const app = workflow.compile()

const result = await app.invoke({ keyDoesNotExist: 'hi' }) // type error here
```

### PR DESCRIPTION
**Pull request by dqbd**
Opting to fix output types in a different PR due to the complexity of `streamMode` 

Closes #628

### CODE REVIEW

**Review by nfcampos — DISMISSED** [review]

File: libs/langgraph/src/graph/graph.ts [hunk 1]
```diff
@@ -441,37 +441,39 @@ export class Graph<
...
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  ConfigurableFieldType & Record<string, any>
+  ConfigurableFieldType & Record<string, any>,
+  Update,
+  State
 > {
   declare NodeType: N;
 
-  declare RunInput: RunInput;
+  declare RunInput: State;
```
Comment (dqbd) [inline libs/langgraph/src/graph/graph.ts:459]: It's a little strange that `RunInput = StateType<SD>`, I think we should change it to Update?

File: libs/langgraph/src/prebuilt/react_agent_executor.ts [hunk 2]
```diff
@@ -160,9 +150,7 @@ export function createReactAgent(
     return { messages: [await modelRunnable.invoke(messages, config)] };
   };
 
-  const workflow = new StateGraph<AgentState>({
-    channels: schema,
-  })
+  const workflow = new StateGraph(MessagesAnnotation)
```
Comment (dqbd) [inline libs/langgraph/src/prebuilt/react_agent_executor.ts:153]: This is done primarily to get the correct input types (based off the `messagesReducer`, but open to different solutions.

File: libs/langgraph/src/tests/pregel.test.ts [hunk 3]
```diff
@@ -3541,6 +3541,7 @@ export function runPregelTests(
           hello: "there",
           bye: "world",
           messages: ["hello"],
+          // @ts-expect-error This should emit a TS error
```
Comment (dqbd) [inline libs/langgraph/src/tests/pregel.test.ts:3544]: We can now actually catch excessive keys!

File: libs/langgraph/src/prebuilt/react_agent_executor.ts (hunk 2 above)
Comment (nfcampos) [reply libs/langgraph/src/prebuilt/react_agent_executor.ts:153]: I think it's good to make this change, the other syntax is deprecated

File: libs/langgraph/src/tests/pregel.test.ts (hunk 3 above)
Comment (nfcampos) [reply libs/langgraph/src/tests/pregel.test.ts:3544]: ha nice!

File: libs/langgraph/src/graph/graph.ts (hunk 1 above)
Comment (nfcampos) [reply libs/langgraph/src/graph/graph.ts:459]: Yea I agree, this looks to be the other way around, no? input should be update, output should be based on stream mode (sometimes update, sometimes state, etc)

**Review by nfcampos — DISMISSED** [review]

**Pull request merged**